### PR TITLE
calc_item_damage_enemy

### PR DESCRIPTION
**Pull request by Brotenko**

### CODE REVIEW

File: src/1A5830.c [hunk 1]
```diff
@@ -72,10 +72,10 @@ void play_hit_sound(Actor* actor, f32 x, f32 y, f32 z, u32 hitSound) {
                     sfx_play_sound_at_position(SOUND_E2, 0, x, y, z);
                     break;
                 case ACTOR_PARTNER:
-                    sfx_play_sound_at_position(SOUND_EA, 0, x, y, z);
+                    sfx_play_sound_at_position(SOUND_FIRE, 0, x, y, z);
```
Comment (ethteck) [inline src/1A5830.c:75]: These sound names don't really mean anything to me. The sound of fire in what context?

File: src/17FEB0.c [hunk 2]
```diff
@@ -37,7 +40,477 @@ s32 calc_item_check_hit(void) {
...
+    
+    targetActor = get_actor(currentTargetID);
+    if (targetActor == NULL) {
+        return 0;
+    }
+
+    targetActorPart = get_actor_part(targetActor, currentTargetPartID);
+    currentTargetID &= 0x700;
+    if (targetActorPart == NULL) {
+        while (1);
```
Comment (ethteck) [inline src/17FEB0.c:83]: this should be PANIC() and the rest of the function doesn't need to be an else, it can just come after.

**Review by ethteck — CHANGES_REQUESTED** [review]
I wish we could get rid of the goto before calling this NON_MATCHING, but Iit's not a dealbreaker I guess. I'll look at this one later

File: src/1A5830.c (hunk 1 above)
Comment (Brotenko) [reply src/1A5830.c:75]: They play accordingly to the dealt damage type "element"

Comment (ethteck) [reply src/1A5830.c:75]: then I think maybe FIRE_DAMAGE and SOUND_ELECTRIC or w/e would be better. Or even SOUND_ELEM_FIRE and SOUND_ELEM_ELECTRIC. just to make it clear that these are battle sounds and not just the sound of fire or electricity in some general sense. if the other one wasn't electricity, my b. I forgot what it was and this tiny diff isn't showing me

File: src/17FEB0.c [hunk 3]
```diff
@@ -40,7 +40,440 @@ s32 calc_item_check_hit(void) {
...
+        if (targetActorPart->eventFlags & ACTOR_EVENT_FLAG_400000) {
+            dispatchEvent = 19;
+            wasStatusInflicted = 1;
+        }
+        if (targetActorPart->eventFlags & ACTOR_EVENT_FLAG_80000) {
+            dispatchEvent = 29;
+        }
+        phi_fp = 0;
+    }
+    if (battleStatus->currentAttackElement & DAMAGE_TYPE_REMOVE_BUFFS && gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE && targetActor->attackBoost <= 0 && targetActor->defenseBoost <= 0 && targetActor->staticStatus == 0 && targetActor->transStatus == 0) {
```
Comment (JoshDuMan) [inline src/17FEB0.c:217]: This line a bit long

File: src/17FEB0.c [hunk 4]
```diff
@@ -40,7 +40,440 @@ s32 calc_item_check_hit(void) {
...
+    }
+    if ((battleStatus->currentAttackElement & DAMAGE_TYPE_REMOVE_BUFFS) == 0) {
+        if (targetActorPart->eventFlags & ACTOR_EVENT_FLAG_ILLUSORY) {
+            return 6;
+        }
+        if (targetActor->transStatus == 14 || targetActorPart->eventFlags & ACTOR_EVENT_FLAG_800 && !(battleStatus->currentAttackElement & DAMAGE_TYPE_QUAKE)) {
+            return 6;
+        }
+    }
+    if (targetActor->stoneStatus == 12) {
```
Comment (nanaian) [inline src/17FEB0.c:116]: This and the other status checks (e.g. transStatus) should use the Statuses enum, i.e.
```suggestion
    if (targetActor->stoneStatus == STATUS_STONE) {
```

File: src/17FEB0.c [hunk 5]
```diff
@@ -40,7 +40,440 @@ s32 calc_item_check_hit(void) {
...
+        return 2;
+    }
+    if (battleStatus->currentAttackElement & DAMAGE_TYPE_2000) {
+        battleStatus->lastAttackDamage = 0;
+        dispatchEvent = 32;
+        phi_fp = 0;
+    }
+    if (gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE) {
+        if (dispatchEvent == 9) {
+            dispatchEvent = 10;
```
Comment (nanaian) [inline src/17FEB0.c:185]: Should use Events enum

**Review by nanaian — CHANGES_REQUESTED** [review]

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 27 functions (+0.06%) on `us`.

✅ This PR fixes 5 warnings!

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀 This PR matches 4 functions (+0.01%) on `us`.

File: include/enums.h [hunk 6]
```diff
@@ -1620,40 +1621,42 @@ enum Areas {
...
-    NPC_FLAG_2000000           = 0x02000000,
+    NPC_FLAG_0                       = 0x00000000,
+    NPC_FLAG_PASSIVE                 = 0x00000001, ///< Collision does not trigger battle
+    NPC_FLAG_2                       = 0x00000002,
+    NPC_FLAG_4                       = 0x00000004,
+    NPC_FLAG_ENABLE_HIT_SCRIPT       = 0x00000008,
+    NPC_FLAG_HAS_SHADOW              = 0x00000010, ///< Set by default and by enable_npc_shadow
+    NPC_FLAG_NO_AI                   = 0x00000020, ///< Disable movement AI and collision (idle animation plays)
+    NPC_FLAG_40                      = 0x00000040,
+    NPC_FLAG_UPSIDE_DOWN             = 0x00000080, ///< Render NPC's upside-down
```
Comment (nanaian) [inline include/enums.h:1634]:
```suggestion
    NPC_FLAG_UPSIDE_DOWN             = 0x00000080, ///< Render NPCs upside-down
```

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

File: src/17FEB0.c [hunk 7]
```diff
@@ -40,7 +40,449 @@ s32 calc_item_check_hit(void) {
...
+            battleStatus->lastAttackDamage = 0;
+            dispatchEvent = EVENT_UNKNOWN_TRIGGER;
+        }
+    } else {
+        targetActor->damageCounter += attackDamage;
+        targetActor->hpChangeCounter -= attackDamage;
+        battleStatus->lastAttackDamage = 0;
+        dispatchEvent = EVENT_HIT_COMBO;
+        phi_fp = 0;
+        if (((targetActorPart->flags & ACTOR_PART_FLAG_2000) == 0) && sp18 == 0 && ((targetActorPart->targetFlags & ACTOR_PART_FLAG_4) == 0)) {
```
Comment (ethteck) [inline src/17FEB0.c:161]: ! for flag checks

File: src/17FEB0.c [hunk 8]
```diff
@@ -40,7 +40,449 @@ s32 calc_item_check_hit(void) {
...
+    sp18 = (targetActorPart->partFlags3 & battleStatus->currentAttackElement) != 0; // TODO 3a8
+    if (targetActorPart->eventFlags & (ACTOR_EVENT_FLAG_ENCHANTED | ACTOR_EVENT_FLAG_80000)) {
+        battleStatus->currentAttackElement &= ~DAMAGE_TYPE_IGNORE_DEFENSE;
+    }
+    targetActorDefense = get_defense(targetActor, targetActorPart->defenseTable, battleStatus->currentAttackElement);
+    if (!(battleStatus->currentAttackElement & DAMAGE_TYPE_IGNORE_DEFENSE)) {
+        targetActorDefense += targetActor->defenseBoost;
+    }
+    attackDamage = battleStatus->currentAttackDamage;
+    if (attackDamage >= 100) {
```
Comment (ethteck) [inline src/17FEB0.c:136]: > 99

**Review by ethteck — CHANGES_REQUESTED** [review]
Before we merge this I'll take a look (tomorrow) and if not, I'm happy to merge after these 2 things are addressed

**Review by ethteck — APPROVED** [review]

**Pull request merged**